sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,13 +349,28 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import sin, sinh
+        arg = self.args[0]
+
+        # `cosh(x)` is real for real and purely imaginary `x`
+        if arg.is_real or arg.is_imaginary:
+            return True
+
+        # cosh(a+ib) = cos(b)*cosh(a) + i*sin(b)*sinh(a)
+        if arg.is_number:
+            re, im = arg.as_real_imag()
+            if (sin(im)*sinh(re)).equals(0):
+                return True
+            return False
 
     def _eval_is_finite(self):
         arg = self.args[0]
         if arg.is_imaginary:
             return True
 
+        if arg.is_real and arg.is_finite:
```

Comment:
cosh is finite for all finite values of the argument. The test `arg.is_imaginary` above is not correct. It probably comes from the observation that cosh is bounded on the imaginary axis.